h2o-algos/src/main/java/hex/glm/GLMModel.java

```diff
@@ -718,6 +728,60 @@ public final double likelihood(double yr, double ym){
         return .5 * deviance(yr,ym);
     }
 
+    // more time-consuming likelihood calculation used for the final scoring
+    public final double likelihood(double w, double yr, double[] ym) {
+      double prediction = ym[0];
+      double probabilityOf1;
+      switch (_family) {
+        case gaussian:
+          return -.5 * (w * Math.pow(yr - prediction , 2) / _dispersion_estimated 
+                  + log(_dispersion_estimated / w) + LOG2PI);
+        case binomial:
+          // if probability is not given, then it is 1.0 if 1 is predicted and 0.0 if 0 is predicted
+          probabilityOf1 = ym.length > 1 ? ym[2] : ym[0]; // probability of 1 equals prediction
+          return w * (yr * log(probabilityOf1) + (1-yr) * log(1 - probabilityOf1)) 
+                  + w * (Gamma.logGamma(2) - Gamma.logGamma(yr + 1) 
+                  - Gamma.logGamma(1 - yr + 1));
+        case quasibinomial:
+          // if probability is not given, then it is 1.0 if 1 is predicted and 0.0 if 0 is predicted
+          probabilityOf1 = ym.length > 1 ? ym[2] : ym[0]; // probability of 1 equals prediction
+          if (yr == prediction)
+            return 0;
+          else if (prediction > 1) // check what are possible values?
+            return -w * (yr * log(probabilityOf1));
+          else
+            return -w * (yr * log(probabilityOf1) + (1 - yr) * log(1 - probabilityOf1));
+        case fractionalbinomial:
+          // if probability is not given, then it is 1.0 if 1 is predicted and 0.0 if 0 is predicted
+          probabilityOf1 = ym.length > 1 ? ym[2] : ym[0]; // probability of 1 equals prediction
+          if (yr == prediction)
+            return 0;
+          return w * ((MathUtils.y_log_y(yr, probabilityOf1)) + MathUtils.y_log_y(1 - yr, 1 - probabilityOf1));
+        case poisson:
+          return w * (yr * log(prediction) - prediction - Gamma.logGamma(yr + 1)); // gamma(n) = (n-1)!
+        case negativebinomial:
+          double invThetaEstimated = 1 / _dispersion_estimated;
+          return yr * log(invThetaEstimated * prediction / w) - (yr + w/invThetaEstimated) * log(1 + invThetaEstimated * prediction / w) 
```

Comment:
Here is our formula for negative binomial:
![image](https://github.com/h2oai/h2o-3/assets/7231712/835d7fbf-61b1-43db-8d3a-a4af88505787)

Here is from outside doc:
![image](https://github.com/h2oai/h2o-3/assets/7231712/0f6b5f40-e150-4ce8-a57b-5219711e5353)

I believe, our theta = k in the other document.  

So, please correct the formua as

yr * log(dispersion_estimated*prediction/w)-(yr+w*invThetaEstimated)*log(1+_dispersion_parameter*prediction/w)+Gamma.logGamma(yr+w*invThetaEstimated)-Gamma.logGamma(yr+1)-Gamma.logGamma(w*invThetaEstiamted)